ydb/core/blobstorage/vdisk/balance/deleter.cpp

```diff
@@ -0,0 +1,194 @@
+#include "defs.h"
+#include "utils.h"
+
+#include <ydb/core/blobstorage/base/vdisk_sync_common.h>
+#include <ydb/core/blobstorage/vdisk/common/vdisk_queues.h>
+#include <ydb/core/blobstorage/vdisk/hullop/blobstorage_hulllog.h>
+#include <ydb/core/blobstorage/vdisk/synclog/blobstorage_synclog_public_events.h>
+#include <ydb/core/blobstorage/vdisk/common/vdisk_private_events.h>
+
+namespace NKikimr {
+namespace NBalancing {
+
+namespace {
+    class TPartsRequester {
+    private:
+        const TActorId NotifyId;
+        const size_t BatchSize;
+        TQueue<TPartInfo> Parts;
+        TReplQuoter::TPtr Quoter;
+        TIntrusivePtr<TBlobStorageGroupInfo> GInfo;
+        TQueueActorMapPtr QueueActorMapPtr;
+
+        TVector<TPartOnMain> Result;
+        ui32 Responses;
+        ui32 ExpectedResponses;
+    public:
+        enum EState {
+            WAITING_RESPONSES,
+            FINISHED,
+        };
+
+        TPartsRequester(TActorId notifyId, size_t batchSize, TQueue<TPartInfo> parts, TReplQuoter::TPtr quoter, TIntrusivePtr<TBlobStorageGroupInfo> gInfo, TQueueActorMapPtr queueActorMapPtr)
+            : NotifyId(notifyId)
+            , BatchSize(batchSize)
+            , Parts(std::move(parts))
+            , Quoter(quoter)
+            , GInfo(gInfo)
+            , QueueActorMapPtr(queueActorMapPtr)
+            , Result(Reserve(BatchSize))
+            , Responses(0)
+            , ExpectedResponses(0)
+        {}
+
+        EState DoJobQuant(const TActorContext &ctx) {
+            if (ExpectedResponses != 0) {
+                return WAITING_RESPONSES;
+            }
+            if (Parts.empty()) {
+                return FINISHED;
+            }
+            Result.resize(Min(Parts.size(), BatchSize));
+            ExpectedResponses = 0;
+            for (ui64 i = 0; i < BatchSize && !Parts.empty(); ++i) {
+                auto item = Parts.front();
+                Parts.pop();
+                Result[i] = TPartOnMain{
+                    .Key=item.Key,
+                    .Ingress=item.Ingress,
+                    .HasOnMain=false
+                };
+
+                auto vDiskId = GetVDiskId(*GInfo, item.Key);
+                auto ev = TEvBlobStorage::TEvVGet::CreateExtremeIndexQuery(
+                    vDiskId, TInstant::Max(), NKikimrBlobStorage::EGetHandleClass::AsyncRead,
+                    TEvBlobStorage::TEvVGet::EFlags::None, i,
+                    {{item.Key.FullID(), 0, 0}}
+                );
+                TReplQuoter::QuoteMessage(
+                    Quoter,
+                    std::make_unique<IEventHandle>(QueueActorMapPtr->at(TVDiskIdShort(vDiskId)), ctx.SelfID, ev.release()),
+                    0
```

Comment:
This is useless unless you specify some valid weight, in bytes. The answer is really expected to consume some resources, isn't it?